### PR TITLE
STORM-913:Use Curator's delete().deletingChildrenIfNeeded() instead of zk/delete-recursive

### PR DESCRIPTION
**Pull request by caofangkun**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -211,7 +211,7 @@
         <guava.version>16.0.1</guava.version>
         <netty.version>3.9.0.Final</netty.version>
         <log4j-over-slf4j.version>1.6.6</log4j-over-slf4j.version>
-        <log4j.version>2.1</log4j.version>
+        <log4j.version>2.3</log4j.version>
```
Comment (HeartSaVioR) [inline pom.xml]: Seems like it is unneeded change.

Comment (caofangkun) [inline pom.xml]: Thank you @HeartSaVioR  
I have removed this unnessary change.

File: storm-core/src/clj/backtype/storm/zookeeper.clj [hunk 2]
```diff
@@ -103,11 +103,14 @@
      (catch Exception e (throw (wrap-in-runtime e))))))
 
 (defnk delete-node
```
Comment (HeartSaVioR) [inline storm-core/src/clj/backtype/storm/zookeeper.clj]: It silently does nothing when path is not exist, but it was throwing NoNodeException when force is false, and default is false.
Only delete-recursive calls delete-node with force = true, and any other usages are using default.

In short, it doesn't show same behavior between two.

**Comment by HeartSaVioR** [comment]
There're many unrelated commits. Could you get rid of these things?

File: storm-core/src/clj/backtype/storm/zookeeper.clj (hunk 2 above)
Comment (HeartSaVioR) [inline storm-core/src/clj/backtype/storm/zookeeper.clj]: Oh, I got it. Whole usages are calling zk/delete-recursive so your changeset makes sense. My mistake.

**Comment by caofangkun** [comment]
@HeartSaVioR 
Thank you for your advise. I have removed unreleated commits and pushed as https://github.com/apache/storm/pull/615/

**Pull request closed without merging**